pkg/performanceprofile/controller/performanceprofile_controller.go

```diff
@@ -174,27 +154,45 @@ func (r *PerformanceProfileReconciler) removeOLMOperator() error {
 
 	for i := range csvs.Items {
 		csv := &csvs.Items[i]
-		if csv.Name == paoCSV {
-			klog.Infof("Removing performance-addon-operator CSV %s", paoCSV)
-			if err := r.Delete(context.TODO(), csv); err != nil {
-				return err
+		deploymentSpecs := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
+		if deploymentSpecs != nil {
+			for _, deployment := range deploymentSpecs {
+				if &deployment.Spec.Template.Spec != nil {
+					containers := deployment.Spec.Template.Spec.Containers
+					if containers != nil {
+						for _, container := range containers {
+							// "performance-addon" will always be a substring of performance-addon-operator URLs.
+							if strings.Contains(container.Image, "performance-addon") {
+								deletedCSVs = append(deletedCSVs, csv.Name)
+								klog.Infof("Removing performance-addon-operator related CSV %s", csv.Name)
+								if err := r.Delete(context.TODO(), csv); err != nil {
```

Comment:
I am unsure if it is a good idea to delete the CSV before the subscription(OLM can re-create it)

Reply:
you have a good point here.